src/widgets/kernel/qtooltip.cpp

```diff
@@ -401,6 +446,20 @@ bool QTipLabel::tipChanged(const QPoint &pos, const QString &text, QObject *o)
        return false;
 }
 
+bool QTipLabel::tipChanged(const QPoint& pos, QLayout* layout, QObject* o)
+{
+    if (QTipLabel::instance->text() != "") //az not sure how to compare layouts.
```

Comment:
Not sure what to compare here, decided to go ahead with checking if a text is displayed, but open to suggestions.

Reply:
Could compare the layout count if there's one set already? Not perfect, but at least it's something. Checking for current text also makes sense. Nice to do both

Reply:
yeah, but I don't want to consider tooltip as the same when layout count matches. My logic here was if we previously displayed a text and now we are about to show layout consider this as a change.
If it was a layout before, check the rectangle. 
I suppose we can say if `layout->count()>0` consider it always as changed? just to be on a safe side? Not sure.

Reply:
I'd be ok with going on the safe side and giving false positives